Android.bp

```diff
@@ -0,0 +1,118 @@
+cc_defaults {
+    name: "maesdk_defaults",
+    cflags: [
+        "-DBUILD_SHARED_LIBS=1",
+        "-DUSE_ROOM=1"
+    ],
+    cppflags: [
+        "-fexceptions",
+        "-Wno-missing-field-initializers",
+        "-Wno-non-virtual-dtor",
+        "-Wno-reorder-ctor",
+        "-Wno-unused-const-variable",
+        "-Wno-unused-parameter",
+        "-Wno-unused-variable"
+    ],
+    rtti: true,
+    vendor_available: true
+}
+
+cc_library_shared {
+    name: "libmaesdk",
+    defaults: ["maesdk_defaults"],
+    srcs: [
+        "lib/api/AllowedLevelsCollection.cpp",
+        "lib/api/AuthTokensController.cpp",
+        "lib/api/ContextFieldsProvider.cpp",
+        "lib/api/CorrelationVector.cpp",
+        "lib/api/DataViewerCollection.cpp",
+        "lib/api/ILogConfiguration.cpp",
+        "lib/api/LogConfiguration.cpp",
+        "lib/api/LogManager.cpp",
+        "lib/api/LogManagerFactory.cpp",
+        "lib/api/LogManagerImpl.cpp",
+        "lib/api/LogManagerProvider.cpp",
+        "lib/api/LogSessionData.cpp",
+        "lib/api/Logger.cpp",
+        "lib/api/capi.cpp",
+        "lib/backoff/IBackoff.cpp",
+        "lib/bond/BondSerializer.cpp",
+        "lib/callbacks/DebugSource.cpp",
+        "lib/compression/HttpDeflateCompression.cpp",
+        "lib/decorators/BaseDecorator.cpp",
+        "lib/filter/EventFilterCollection.cpp",
+        "lib/http/HttpClientFactory.cpp",
+        "lib/http/HttpClientManager.cpp",
+        "lib/http/HttpRequestEncoder.cpp",
+        "lib/http/HttpResponseDecoder.cpp",
+        "lib/jni/JniConvertors.cpp",
+        "lib/jni/LogManager_jni.cpp",
+        "lib/jni/Logger_jni.cpp",
+        "lib/jni/SemanticContext_jni.cpp",
+        "lib/jni/Utils_jni.cpp",
+        "lib/offline/MemoryStorage.cpp",
+        "lib/offline/LogSessionDataProvider.cpp",
+        "lib/offline/OfflineStorageFactory.cpp",
+        "lib/offline/OfflineStorageHandler.cpp",
+        "lib/offline/StorageObserver.cpp",
+        "lib/packager/BondSplicer.cpp",
+        "lib/packager/Packager.cpp",
+        "lib/pal/InformationProviderImpl.cpp",
+        "lib/pal/PAL.cpp",
+        "lib/pal/TaskDispatcher_CAPI.cpp",
+        "lib/pal/WorkerThread.cpp",
+        "lib/pal/posix/DeviceInformationImpl_Android.cpp",
+        "lib/pal/posix/NetworkInformationImpl_Android.cpp",
+        "lib/pal/posix/SystemInformationImpl_Android.cpp",
+        "lib/pal/posix/sysinfo_sources.cpp",
+        "lib/stats/MetaStats.cpp",
+        "lib/stats/Statistics.cpp",
+        "lib/system/EventProperties.cpp",
+        "lib/system/EventProperty.cpp",
+        "lib/system/TelemetrySystem.cpp",
+        "lib/tpm/DeviceStateHandler.cpp",
+        "lib/tpm/TransmissionPolicyManager.cpp",
+        "lib/tpm/TransmitProfiles.cpp",
+        "lib/utils/FileUtils.cpp",
+        "lib/utils/StringUtils.cpp",
+        "lib/utils/ZlibUtils.cpp",
+        "lib/utils/Utils.cpp",
+        "lib/offline/OfflineStorage_Room.cpp",
+        "lib/http/HttpClient_Android.cpp"
```

Comment:
How is this list sourced? How do we ensure that someone doesn't miss updating this list when a file is updated? Can this file be validated via Actions/CIs?

Reply:
This part of the .bp file is for creating the native library, so we borrowed the entries from what CMake is doing. Specifically, these entries are sourced from the [CMakeList.txt](https://github.com/microsoft/cpp_client_telemetry/blob/327c219d0b0b0f83ed2b8f501d1a3c8c95c12433/lib/android_build/maesdk/src/main/cpp/CMakeLists.txt#L22) file. We also add an additional two entries at the bottom of this list (OfflineStorage_Room and HttpClient_Android). These are conditionally added in the CMake file, but we add them to our list here since Android.bp ([Soong](https://android.googlesource.com/platform/build/soong/) build system) does not support conditionals or control flow.

Since this list is already maintained in that Cmake txt file, it would be great if we could somehow point to that without having to re-create it here in this build file. I don't know of a solution to that at this time.

I need to investigate what options we have for Actions/CI checks.

Reply:
Sounds good. In that case, I'd suggest at least having a comment in the CMakeList.txt to point authors towards updating this file as well. Preferably if we can have CI validation, that'll be best, but if we can alternatively have some script that does a quick validation to ensure the lists are in sync, it will ensure you don't have to go debug weird build breaks you may run into.

I'm satisfied with this answer and am signing off with the understanding you'll investigate the Action/CI and consider adding a comment to CMakeList.txt.